bin/stringfuzzx

```diff
@@ -51,25 +65,53 @@ def main():
         default = SMT_25_STRING,
         help    = 'output language (default: {})'.format(SMT_25_STRING)
     )
-    parser.add_argument(
-        'transformer',
+    global_parser.add_argument(
+        'transformers',
         type    = str,
         choices = TRANSFORMERS.keys(),
-        help    = 'transformation to perform'
+        nargs   = "+",
+        help    = 'transformations to perform'
+    )
+
+    seed_group = global_parser.add_mutually_exclusive_group()
+    seed_group.add_argument(
+        '--seed',
+        '-s',
+        dest    = 'seed',
+        metavar = 'S',
+        type    = int,
+        default = DEFAULT_SEED,
+        help    = 'seed for random number generator (default: {})'.format(DEFAULT_SEED)
+    )
+    seed_group.add_argument(
+        '--random',
+        '-r',
+        dest    = 'random',
+        action  = 'store_true',
+        default = DEFAULT_RANDOM,
+        help    = 'seed the random number generator with the current time (default: {})'.format(DEFAULT_RANDOM)
     )
 
     # parse args
-    args = parser.parse_args()
+    args = global_parser.parse_args()
+
+    # seed the RNG
+    if args.random is True:
+        random.seed()
+    else:
+        random.seed(args.seed)
 
     # get transformer
-    transformer = TRANSFORMERS[args.transformer]
+    transformers = [TRANSFORMERS[a] for a in args.transformers]
```

Comment:
I really like this idea of chaining transformers. My one concern is argument-passing. How will we be able to parametrise each transformer using this pattern?

As an alternative, what do you think of keeping the one-transformer-at-a-time pattern as before, but chaining them using shell pipes, like so: `cat problem.smt25 | stringfuzzx rotate | stringfuzzx unprintable`?

Reply:
Argument passing is definitely a concern for chaining, I'm not sure what the best way to handle that would be. 

The transformers I've been making are reasonably simple, so I think chaining will be a regular use case. I worry that parsing and writing the problems as is done with shell pipes will be a bit inefficient.

I am also open to the idea of making more powerful transformers that don't need to be chained so frequently. Maybe that is more useful than a few small transformers anyway.

What do you think?

Reply:
I think doing them one at a time and handling arguments the same way as `stringfuzzg` is fine. How inefficient, in extra run time, do you foresee shell chaining might be?